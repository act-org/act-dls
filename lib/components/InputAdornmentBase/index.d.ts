/**
 * @prettier
 */
import * as React from 'react';
import { InputAdornmentProps } from '@material-ui/core';
export declare type Props = InputAdornmentProps;
declare const InputAdornmentBase: React.FC<Props>;
export default InputAdornmentBase;
//# sourceMappingURL=index.d.ts.map