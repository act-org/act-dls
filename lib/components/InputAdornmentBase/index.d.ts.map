{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/InputAdornmentBase/index.tsx"],"names":[],"mappings":"AAAA;;GAEG;AAEH,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAC/B,OAAO,EAAkB,mBAAmB,EAAE,MAAM,mBAAmB,CAAC;AAExE,oBAAY,KAAK,GAAG,mBAAmB,CAAC;AAExC,QAAA,MAAM,kBAAkB,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAEmB,CAAC;AAE5D,eAAe,kBAAkB,CAAC"}