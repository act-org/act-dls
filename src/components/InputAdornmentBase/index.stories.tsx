/**
 * @prettier
 */

/* eslint-disable sort-keys */

import * as React from 'react';
import { Plus } from 'mdi-material-ui';
import { select } from '@storybook/addon-knobs';

import InputAdornmentBase from '.';

export const Base = (): React.ReactElement<any> => (
  <InputAdornmentBase
    position={select(
      'Position',
      {
        Start: 'start',
        End: 'end',
      },
      'start',
    )}
  >
    <Plus />
  </InputAdornmentBase>
);

export default {
  component: InputAdornmentBase,
  parameters: {
    info: {
      text: 'This is the base variant of the InputAdornment component.',
    },
  },
  title: 'Atoms/InputAdornment',
};
