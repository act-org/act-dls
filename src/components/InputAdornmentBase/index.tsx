/**
 * @prettier
 */

import * as React from 'react';
import { InputAdornment, InputAdornmentProps } from '@material-ui/core';

export type Props = InputAdornmentProps;

const InputAdornmentBase: React.FC<Props> = (
  props: Props,
): React.ReactElement<any> => <InputAdornment {...props} />;

export default InputAdornmentBase;
