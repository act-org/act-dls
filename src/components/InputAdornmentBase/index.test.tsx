/**
 * @prettier
 */

import * as React from 'react';

import render from '~/helpers/test/render';

import InputAdornmentBase from '.';

describe('InputAdornmentBase', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <InputAdornmentBase position="start">InputAdornment</InputAdornmentBase>,
    );

    expect(container).toMatchSnapshot();
  });
});
