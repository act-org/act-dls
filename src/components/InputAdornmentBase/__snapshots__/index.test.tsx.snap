// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputAdornmentBase matches the snapshot 1`] = `
<div>
  <div
    class="MuiInputAdornment-root MuiInputAdornment-positionStart"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      InputAdornment
    </p>
  </div>
</div>
`;
